python/src/boombox/boombox.py

```diff
@@ -0,0 +1,471 @@
+
+    def __exit__(self, *_) -> None:
+        self.close()
+
+    @override
+    def handle_one_inbox_message(self, msg: Any) -> None:
+        """:meta private:"""
+        if self._response_received is not None:
+            self._response_received.set_result(msg)
+
+    @override
+    def exit(self, reason: Any = None) -> None:
+        """:meta private:"""
+        self._finished.set_result(None)
+        super().exit(reason)
+
+    def _call(self, request: Any) -> Any:
+        message = (
+            Atom("call"),
+            (Atom(self._process_name), Atom(self.node_name_)),
+            request,
+        )
+
+        self.get_node().send_nowait(
+            sender=self, receiver=self._receiver, message=message
+        )
+        self._response_received = self.get_node().get_loop().create_future()
+        return self.get_node().get_loop().run_until_complete(self._response_received)
+
+    def _dtype_to_sample_format(
+        self, dtype: np.dtype
+    ) -> tuple[AudioSampleFormat, np.dtype]:
+        match dtype.kind:
+            case "i":
+                data_type = "s"
+            case "u":
+                data_type = "u"
+            case "f":
+                data_type = "f"
+            case other:
+                warnings.warn(
+                    f"Arrays of dtype.kind == {other} not allowed, supported kinds are 'i', 'u', 'f', casting to 'f'."
+                )
+                data_type = "f"
+
+        match dtype.itemsize:
+            case 1 | 2 | 4 | 8:
+                bit_size = str(dtype.itemsize * 8)
+            case other:
+                warnings.warn(
+                    f"Item size {dtype.itemsize} not allowed, supported item sizes are 1, 2, 4 and 8, casting to 4"
+                )
+                bit_size = "32"
+
+        match dtype.byteorder:
+            case "<":
+                endian = "le"
+            case ">":
+                endian = "be"
+            case "=":
+                endian = "le" if sys.byteorder == "little" else "be"
+            case "|":
+                endian = ""
+
+        sample_format = data_type + bit_size + endian
+        # trick to ensure that sample_format is of AudioSampleFormat type
+        audio_sample_formats: tuple[AudioSampleFormat, ...] = get_args(
+            AudioSampleFormat
+        )
+        assert sample_format in audio_sample_formats
+
+        new_dtype = self._sample_format_to_dtype(sample_format)
+
+        return sample_format, new_dtype
+
+    def _sample_format_to_dtype(self, sample_format: AudioSampleFormat) -> np.dtype:
+        type_mapping = {"s": "i", "u": "u", "f": "f"}
+
+        if len(sample_format) == 2:  # Handle 8-bit case (without endian)
+            data_type = sample_format[0]
+            dtype_str = type_mapping[data_type] + "1"
+        else:
+            data_type, bit_size, endian = (
+                sample_format[0],
+                sample_format[1:-2],
+                sample_format[-2:],
+            )
```

Comment:
[NIT] I don't think we benefit anyhow from setting it as a tuple :D 
```suggestion
            data_type = sample_format[0]
            bit_size = sample_format[1:-2]
            endian = sample_format[-2:]
```